rasa/core/policies/rule_policy.py

```diff
@@ -74,6 +80,154 @@ def _create_feature_key(self, states: List[Dict]) -> Text:
 
         return feature_str
 
+    @staticmethod
+    def _get_active_form_name(state: Dict[Text, float]) -> Optional[Text]:
+        found_forms = [
+            state_name[len(ACTIVE_FORM_PREFIX) :]
+            for state_name, prob in state.items()
+            if ACTIVE_FORM_PREFIX in state_name
+            and state_name != ACTIVE_FORM_PREFIX + "None"
+            and prob > 0
+        ]
```

Comment:
let's use a generator as we don't have to process every state once we have found a form
```suggestion
                found_forms = next(
            (
                state_name[len(ACTIVE_FORM_PREFIX) :]
                for state_name, prob in state.items()
                if ACTIVE_FORM_PREFIX in state_name
                and state_name != ACTIVE_FORM_PREFIX + "None"
                and prob > 0
            ),
            None,
        )
```